src/core/moves/proposal/tree/NodeRateTimeSlideUniformProposal.cpp

```diff
@@ -117,14 +117,31 @@ double NodeRateTimeSlideUniformProposal::doProposal( void )
     {
         return 0.0;
     }
+    
+    // check that there is at least one node which is not the root, a tip, or the parent of a SA
+    std::vector<TopologyNode*> eligible_nodes;
+    
+    for (auto& to_check: tau.getNodes())
+    {
+        if ( !to_check->isRoot() && !to_check->isTip() && !to_check->isSampledAncestorParent() )
+        {
+            eligible_nodes.push_back( to_check );
+        }
+    }
+    
+    if (eligible_nodes.size() == 0)
+    {
+        std::cerr << "mvNodeRateTimeSlideUniform has no effect; the tree only contains the root, tips, and sampled ancestors." << std::endl;
+        return RbConstants::Double::neginf;
+    }
 
     // pick a random node which is not the root, a tip, or the parent of a SA
     TopologyNode* node;
     do {
         double u = rng->uniform01();
         size_t index = size_t( std::floor(tau.getNumberOfNodes() * u) );
         node = &tau.getNode(index);
-    } while ( node->isRoot() || node->isTip() || node -> isSampledAncestorParent() );
+    } while ( node->isRoot() || node->isTip() || node->isSampledAncestorParent() );
```

Comment:
If we've gone to all the work to put the eligible nodes into a vector, then we can just select an index from the vector instead of doing this weird while loop.

Reply:
Makes sense!